### PR TITLE
Submission for Takeaway Challenge

### PR DESCRIPTION
**Pull request by katylouise**
Solution so far - need to do some refactoring and have a go at the extension.

### CODE REVIEW

File: lib/takeaway.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+require 'twilio-ruby'
+
+class TakeAway
+
+  attr_reader :menu
+  attr_accessor :order, :prices
+
+  def initialize
```
Comment (zdajani) [inline lib/takeaway.rb]: initialise with a menu, which would remove the need for the next two methods

File: lib/takeaway.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  def add_menu(menu)
+    @menu = menu.dishes
+  end
+
+  def show_menu
+    menu
+  end
+
+  def place_order(items = {})
```
Comment (zdajani) [inline lib/takeaway.rb]: separate items from quantity so you can add to the quantity, without having to add the same item twice

File: lib/takeaway.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  def total_cost_per_item(price, quantity)
+    price_in_pence = ((price * 100).round * quantity)
+    prices << price_in_pence
+    "%.2f" % (price_in_pence / 100.0)
+  end
+
+  def show_order
+    order_output = print_order_items
+    order_output += "Total: £#{grand_total}"
+    order_output
```
Comment (zdajani) [inline lib/takeaway.rb]: you can remove this line

**Comment by zdajani** [comment]
Really good! I would just suggest making a module for the calculations of the order and keep the takeaway class just for ordering. Also think about separating the Twilio part into another module or class.

File: lib/takeaway.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      order["#{dish} x #{quantity}"] = total_cost_per_item(menu[dish], quantity)
+    end
+  end
+
+  def grand_total
+    total_in_pence = prices.inject(:+)
+    "%.2f" % (total_in_pence / 100.0)
+  end
+
+  def total_cost_per_item(price, quantity)
```
Comment (zdajani) [inline lib/takeaway.rb]: check if you can delete by quantity

**Comment by stephenlloyd** [comment]
:checkered_flag:

**Pull request closed without merging**